fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java

```diff
@@ -6546,6 +6547,35 @@ public void changeDb(ConnectContext ctx, String qualifiedDb) throws DdlException
         ctx.setDatabase(qualifiedDb);
     }
 
+    // Change current catalog and database of this session.
+    // We can support 'USE CATALOG.DB'
+    public void changeCatalogDb(ConnectContext ctx, String identifier) throws DdlException {
+        String catalogName = ctx.getCurrentCatalog();
+
+        String[] parts = identifier.split("\\.");
+        if (parts.length != 1 && parts.length != 2) {
+            ErrorReport.reportDdlException(ErrorCode.ERR_BAD_CATALOG_ERROR, identifier);
+
+        } else if (parts.length == 1) {
+            String dbName = catalogName.equals(InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME) ?
+                    ClusterNamespace.getFullName(ctx.getClusterName(), identifier) : identifier;
+            changeDb(ctx, dbName);
+
+        } else {
+            String newCatalogName = parts[0];
+            if (GlobalStateMgr.getCurrentState().getCatalogMgr().catalogExists(newCatalogName)) {
```

Comment:
There is no internal catalog instance in catalogMgr. should we add it in catalogMgr?